### PR TITLE
Enable CSRF token validation

### PR DESCRIPTION
**Pull request by pjanik**
Cross site forgery protection has been turned on (including Cucumber tests).

I was testing or looking for:

#### jQuery Ajax calls: $.ajax / $.post
I fixed them by using $.ajaxSetup in `jquery_ajax_csrf_token.js`. It automatically adds X-CSRF-Token header to every ajax request.

#### Prototype.js Ajax.Request, Ajax.Updater:
It was already handled by `rails.js` => X-CSRF-Token header is added to every request.

#### Other JavaScript libraries issuing post requests
The only thing that came to my mind was Angular.JS and collaboration setup. I’ve handled that in AngularJS setup code.

#### Custom forms created using `%form` tag (:method => :POST)
Rails helpers add CSRS token automatically, but forms created manually need to be handled separately. I've searched form them and manually added `hidden_field_tag :authenticity_token, form_authenticity_token`

#### new API (`app/controllers/api/api_controller.rb`)
External report uses token authentication, so that's fine. Any other sites that might be using cookie-based auth? We can consider disabling CSRF check for this API, but it would be nice to have it enabled, so right approach (token-based auth) is automatically enforced.

#### External report
Token authentication, I've tested it and it works. 
Note: CSRF token validation warning is displayed, but not causing any issues. I’ve double checked that user is not logged out from Portal and there are no unwanted effects. I'm not sure how to get rid of the warning, except from disabling CSRF token check for this action. I was thinking about that, but can't find a solution. In the perfect scenario, CSRF token should be checked only when cookie-based auth is being used, but it doesn't seem to work like this - it's just a controller's `before_filter`.

#### Activity publication
Token authentication, I've tested it and it works. 
Note: CSRF token validation warning is displayed.

#### Answers LARA -> Portal
Token authentication, I've tested it and it works. 
Note: CSRF token validation warning is displayed.

#### Other Dataservice:: controllers
No idea about them, but I guess JNLP activities can't use cookies anyway, so there needs to be some kind of authentication that works. Geniverse is using some of those controllers, but they inherit form `ActionController::Metal` which doesn't have CSRF token check enabled.

#### Random elements
I've tested search, assign to class, some forms in admin panel (permission search), etc. They should be all covered by use cases listed above anyway or Rails helpers.

#### Geniverse and Genigames

Some of the POST / PUT requests: 

POST /remote_login
POST /remote_logout
POST /dataservice/bucket_loggers/name/<username>/bucket_log_items.bundle
POST /geniverse/users
PUT /geniverse/users/27045.json
and more /geniverse/... namespaced actions

/geniverse - fixed in https://github.com/concord-consortium/geniverse-portal-integration/pull/1
/remote_login / logout - fixed in https://github.com/concord-consortium/cc_portal_remote_auth/pull/1

Also, I've done some testing and it seems to work fine.

### CODE REVIEW

File: app/assets/javascripts/angular/portal_app.js.coffee [hunk 1]
```diff
@@ -1 +1,4 @@
-angular.module('cc-portal', ['ccCollaboration'])
+angular.module('cc-portal', ['ccCollaboration']).
+  config ["$httpProvider", ($httpProvider) ->
+    $httpProvider.defaults.headers.common['X-CSRF-Token'] = jQuery('meta[name=csrf-token]').attr('content')
+  ]
```
Comment (scytacki) [inline app/assets/javascripts/angular/portal_app.js.coffee]: What is angular used for now?

Comment (pjanik) [inline app/assets/javascripts/angular/portal_app.js.coffee]: There's only one thing left - a collaboration setup dialog.

**Comment by pjanik** [comment]
@scytacki, thanks, good catch. I've checked custom home pages and project pages. I think Ethan is using our libraries (jQuery) and some forms are provided by Portal (e.g. signup). But I found that there's a problem with custom login form, as the authenticity token was missing.

I was thinking about fixing manually all the project pages, but finally I updated Portal code:
https://github.com/concord-consortium/rigse/pull/292/commits/413ca60af0475d6e041030ca8a2f432334327d82

This automatically fixes all the custom pages and Ethan won't have to remember about it.

**Comment by scytacki** [comment]
I'm guessing we still missed something, but I'm going to merge it and we'll see what Evangline and other find as they use this on staging.

**Pull request merged**